pkg/controllers/grafana-operator/controller.go

```diff
@@ -276,6 +291,42 @@ func (r *reconciler) reconcileSubscription(ctx context.Context) reconcileResult
 	return updationResult(err)
 }
 
+func (r *reconciler) approveInstallPlan(ctx context.Context) reconcileResult {
+	subscriptionKey := types.NamespacedName{
+		Name:      subscriptionName,
+		Namespace: Namespace,
+	}
+	var subscription v1alpha1.Subscription
+	if err := r.k8sClient.Get(ctx, subscriptionKey, &subscription); err != nil {
+		return updationResult(err)
+	}
+
+	installPlanRef := subscription.Status.InstallPlanRef
+	if installPlanRef == nil {
+		return reconcileResult{
+			stop: true,
+		}
+	}
+
+	installPlanKey := types.NamespacedName{
+		Namespace: Namespace,
+		Name:      installPlanRef.Name,
+	}
+	var installPlan v1alpha1.InstallPlan
+	err := r.k8sClient.Get(ctx, installPlanKey, &installPlan)
+	if err != nil {
+		return reconcileError(err)
+	}
+
+	// Only approve install plans for the specified version
+	if installPlan.Spec.ClusterServiceVersionNames[0] != grafanaCSV {
+		return updationResult(nil)
```

Comment:
did you want to `return next()` ?

Reply:
I don't know what these terms mean :)

Reply:
I just want to return a no-op basically

Reply:
😂
`next` moves to the next step in the reconcile loop
`end` terminates the reconcile loop

Reply:
Ok, updated